<?xml version="1.0" encoding="UTF-8" ?>
<launch>
    <arg name="config" default="''"/>
    <arg name="rviz" default="true"/>
    <arg name="bag" default="$(dirname)/../data/bags/leica_ouster/small_board/depth_correction_black_board_ouster_leica_2023-02-02-14-50-38.bag"/>
<!--    <arg name="bag" default="$(dirname)/../data/bags/leica_ouster/small_board/depth_correction_white_board_ouster_leica_2023-02-02-13-44-08.bag"/>-->
    <!-- Cloud filter -->
    <arg name="min_depth" default="1.0"/>
    <arg name="max_depth" default="10.0"/>
    <arg name="grid_res" default="0.1"/>
    <!-- Depth correction -->
    <arg name="depth_correction" default="false"/>
    <arg name="nn_k" default="0"/>
    <arg name="nn_r" default="0.25"/>
    <arg name="shadow_neighborhood_angle" default="0.017453"/>
    <arg name="shadow_angle_bounds" default="[0.087266, .nan]"/>
    <arg name="eigenvalue_bounds" default="[[0, -.inf, 0.0025], [1, 0.0025, .inf]]"/>
    <arg name="model_class" default="BaseModel"/>
    <arg name="model_args" default="''"/>
    <arg name="model_kwargs" default="{}"/>
    <arg name="model_state_dict" default="''"/>
    <arg name="device" default="cpu"/>
    <!-- Bag recording -->
    <arg name="record" default="false"/>
    <arg name="recorded_bag" default=""/>

    <!-- Robot data: from bag file -->
    <param name="use_sim_time" value="true"/>
    <node name="rosbag_play" pkg="rosbag" type="play" args="--start 30.0 --delay 3.0 --rate 2.0 $(arg bag)">
        <remap from="ouster/points" to="cloud"/>
    </node>

    <!-- Leica data postprocessing -->
    <node pkg="depth_correction" type="leica_ouster_postproc" name="leica_ouster_postproc" output="screen">
        <rosparam subst_value="true">
            depth_correction: $(arg depth_correction)
        </rosparam>
<!--        <remap from="cloud" to="$(eval 'corrected_cloud' if depth_correction else 'filtered_cloud')"/>-->
        <remap from="cloud" to="cloud"/>
        <remap from="crystal_points" to="total_station_driver/ts_points"/>
    </node>

    <!-- RVIZ -->
    <include if="$(arg rviz)" file="$(dirname)/rviz.launch">
        <arg name="config" value="$(dirname)/../config/leica_ouster.rviz"/>
    </include>
</launch>
